python/cudf/cudf/core/udf/__init__.py

```diff
@@ -1 +1,61 @@
-from . import typing, lowering
+# Copyright (c) 2022, NVIDIA CORPORATION.
+from . import masked_typing, masked_lowering
+from numba import cuda
+from numba import types
+from numba.cuda.cudaimpl import (
+    lower as cuda_lower,
+    registry as cuda_lowering_registry,
+)
+from cudf.core.udf import api
+from cudf.core.udf import utils
+from cudf.core.udf import row_function
+from cudf.core.dtypes import dtype
+import numpy as np
+
+units = ["ns", "ms", "us", "s"]
+datetime_cases = {types.NPDatetime(u) for u in units}
+timedelta_cases = {types.NPTimedelta(u) for u in units}
+
+
+supported_masked_types = (
+    types.integer_domain
+    | types.real_domain
+    | datetime_cases
+    | timedelta_cases
+    | {types.boolean}
+)
+
+_STRING_UDFS_ENABLED = False
+try:
+    import strings_udf
+
+    if strings_udf.ENABLED:
+        from . import strings_typing
+        from . import strings_lowering
+        from strings_udf import ptxpath
+        from strings_udf._typing import string_view, str_view_arg_handler
+        from strings_udf._lib.cudf_jit_udf import to_string_view_array
+
+        # add an overload of MaskedType.__init__(string_view, bool)
+        cuda_lower(api.Masked, strings_typing.string_view, types.boolean)(
+            masked_lowering.masked_constructor
+        )
+
+        # add an overload of pack_return(string_view)
+        cuda_lower(api.pack_return, strings_typing.string_view)(
+            masked_lowering.pack_return_scalar_impl
+        )
+
+        supported_masked_types |= {strings_typing.string_view}
+        utils.launch_arg_getters[dtype("O")] = to_string_view_array
+        utils.masked_array_types[dtype("O")] = string_view
+        utils.files.append(ptxpath)
+        utils.arg_handlers.append(str_view_arg_handler)
+        row_function.itemsizes[dtype("O")] = string_view.size_bytes
+
+        _STRING_UDFS_ENABLED = True
+except ImportError:
+    # allow cuDF to work without strings_udf
```

Comment:
TL;DR, the plan is to keep `strings_udf` as a separately installable bolt-on while we make progress on functions that return strings and some of their unique technical challenges. These functions are built around the c++ class `dstring` which we [decided to keep separate](https://github.com/rapidsai/cudf/pull/10686) from libcudf for now. As long as this c++ component need be separate from libcudf, it seems like it is not useful to provide the python components to cuDF. 

Technically, read-only functions only depend only `cudf::string_view` and thus only use c++ code that is already included in libcudf (modulo the shim layer functions), and for those one could reconfigure things such that a separate package is not needed. But in my opinion keeping read-only functions and functions that return new strings together is a cleaner architecture and allows us to make changes that affect all string related functions cleanly, such as changes to the way our shim functions are compiled. It also prevents us from having to link together even more PTX files.

Ultimately following the line of logic in the linked PR, I could see functions that return strings being built around a fully featured `cuda::std::string` class, at which point we could integrate everything since that's already a dependency of libcudf.